src/GameLogic/Party.cs

```diff
@@ -232,10 +232,13 @@ private async ValueTask<int> InternalDistributeExperienceAfterKillAsync(IAttacka
         var randomizedTotalExperiencePerLevel = randomizedTotalExperience / totalLevel;
         foreach (var player in this._distributionList)
         {
-            if (player.SelectedCharacter?.CharacterClass?.IsMasterClass ?? false)
+            if ((short)player.Attributes![Stats.Level] == this._distributionList.First().GameContext.Configuration.MaximumLevel)
```

Comment:
Instead of ```this._distributionList.First()``` you could simply use ```player```.

Reply:
Thank you! I just added that change and is working correctly